web/src/features/query/types.ts

```diff
@@ -0,0 +1,102 @@
+import { z } from "zod";
+import { singleFilter } from "@langfuse/shared";
+
+export type ViewDeclarationType = z.infer<typeof viewDeclaration>;
+export type DimensionsDeclarationType = z.infer<
+  typeof viewDeclaration
+>["dimensions"];
+
+export const viewDeclaration = z.object({
+  name: z.string(),
+  // This is the basic statement that we query from. Usually, this should be the view_name + FINAL or a more complex subquery.
+  baseCte: z.string(),
+  dimensions: z.record(
+    z.object({
+      sql: z.string(),
+      alias: z.string().optional(),
+      type: z.enum(["string", "number", "bool"]),
```

Comment:
Not used right now, but should become relevant once we have the frontend builder to highlight applicable filter options.